### PR TITLE
feat(cdk): Add Dockerfile

### LINKED ISSUE #1302: Publish public Docker image containing the CDK CLI
Some of the newer CI/CD tools out there (Bitbucket Pipelines, GitHub Actions) use Docker images as the environment for each step of the pipeline.  Packaging the CDK CLI as a public Docker image would be a convenient way to integrate releasing CDK-based apps into these types of CI/CD tools.

### PR DESCRIPTION
**Pull request by jonathanmorley**
fixes #1302

This PR should be followed up with a Dockerhub automated build configured to build from git tags

----

### Pull Request Checklist

* [ ] Testing
  - Unit test added
  - __CLI change?:__ manually run [integration tests](https://github.com/awslabs/aws-cdk/blob/master/packages/aws-cdk/integ-tests/test.sh) and paste output as a PR comment
  - __cdk-init template change?:__ coordinated update of integration tests with team
* [ ] Docs
  - __jsdocs__: All public APIs documented
  - __README__: README and/or documentation topic updated
* [ ] Title and Description
  - __Change type__: title prefixed with **fix**, **feat** will appear in changelog
  - __Title__: use lower-case and doesn't end with a period
  - __Breaking?__: last paragraph: "BREAKING CHANGE: <describe what changed + link for details>"
  - __Issues__: Indicate issues fixed via: "**Fixes #xxx**" or "**Closes #xxx**"
* [ ] Sensitive Modules (requires 2 PR approvers)
  - IAM Policy Document (in @aws-cdk/aws-iam)
  - EC2 Security Groups and ACLs (in @aws-cdk/aws-ec2)
  - Grant APIs (only if not based on official documentation with a reference)

----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

### CODE REVIEW

**Comment by jonathanmorley** [comment]
The use-cases are mainly around using the CDK in environments where running tools in docker is preferred over installing and running through npm. This gives an easier and more official way to run the CDK in that way.

As examples of this there are a number of CI platforms that either have first-class support (e.g. Jenkins pipelines), or only support (e.g. GitHub Actions), for running via a docker image.

**Comment by Doug-AWS** [comment]
This is a great addition to the CDK. I'll add a blurb in the guide once we merge, and make it public once it's in a release.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+FROM node:8.15-alpine
+
+WORKDIR /app
+
+RUN apk add --update \
+    bash \
+    git \
+    rsync \
+    zip \
+    python3 \
```
Comment (RomainMuller) [inline Dockerfile]: What do we need python3 for? o_O

Comment (jonathanmorley) [reply Dockerfile]: They are used during testing, see https://github.com/awslabs/aws-cdk/blob/master/packages/%40aws-cdk/aws-s3-deployment/lambda/test.sh#L19-L22

**Pull request merged**